sdk/python/tests/compiler/component_builder_test.py

```diff
@@ -49,6 +52,56 @@ def test_upload_gcs_path(self):
     os.remove(temp_file)
     os.remove(temp_downloaded_file)
 
+class TestDependencyVersion(unittest.TestCase):
+
+  def test_version(self):
+    """ test version overrides min_version and max_version """
+    version = DependencyVersion(version='0.3.0', min_version='0.1.0', max_version='0.4.0')
+    self.assertTrue(version.min_version == '0.3.0')
+    self.assertTrue(version.max_version == '0.3.0')
+    self.assertTrue(version.has_versions())
+    self.assertTrue(version.has_min_version())
+    self.assertTrue(version.has_max_version())
+
+  def test_minmax_version(self):
+    """ test if min_version and max_version are configured when version is not given """
+    version = DependencyVersion(min_version='0.1.0', max_version='0.4.0')
+    self.assertTrue(version.min_version == '0.1.0')
+    self.assertTrue(version.max_version == '0.4.0')
+    self.assertTrue(version.has_versions())
+    self.assertTrue(version.has_min_version())
+    self.assertTrue(version.has_max_version())
```

Comment:
Maybe we should just check string rendering of the version?

Reply:
I'm just not sure a test like this is useful. It puts some data in and immediately checks it. It would be a bit more interesting to check something like `dependency.to_requirements_line()`

Reply:
I think tests like these still bring some benefits. It guarantees the expected behavior from the VersionDependency class is as expected by other functions. Potentially, we want to add new members and maybe change the behavior for has_versions(), having the unit tests might help.

How about I removing the basic ones such as has_min_version, has_max_version, whose behavior is never going to be changed?

Reply:
The reason I'm asking for a more thorough testing here is that your code has a bug and your tests failed to catch it. I'm concerned about this.

Reply:
Can you be more specific about the bug? Thanks

Reply:
I just want to address the test design so that tests make it harder for bugs to sneak in.

I think that these tests serve little purpose, but you might want more tests in `TestDependencyHelper` class.

Reply:
I see your points. I will delete some checks and add tests for DependencyHelper

Reply:
Thanks. Done.